src/operation/watch/mod.rs

```diff
@@ -0,0 +1,104 @@
+            None => doc.insert("$changeStream", doc! {}),
+        };
+        let mut aggregate_pipeline = Vec::new();
+        let pipeline: Vec<Document> = pipeline.into_iter().collect();
+        let target: ChangeStreamTarget = target.into();
+        match &target {
+            ChangeStreamTarget::Collection(collection) => {
+                let aggregate_target = AggregateTarget::Collection(collection.namespace());
+                aggregate_pipeline.push(doc);
+                aggregate_pipeline.extend(pipeline.clone());
+                let aggregate_operation = Aggregate::new(
+                    aggregate_target,
+                    aggregate_pipeline,
+                    options.clone().map(|opts| opts.get_aggregation_options()),
+                );
+                Self {
+                    target,
+                    pipeline: pipeline.into_iter().collect(),
+                    options,
+                    aggregate_operation,
+                }
+            }
+            ChangeStreamTarget::Database(database) => {
+                let aggregate_target = AggregateTarget::Database(database.name().to_string());
+                aggregate_pipeline.push(doc);
+                aggregate_pipeline.extend(pipeline.clone());
+                let aggregate_operation = Aggregate::new(
+                    aggregate_target,
+                    aggregate_pipeline,
+                    options.clone().map(|opts| opts.get_aggregation_options()),
+                );
+                Self {
+                    target,
+                    pipeline: pipeline.into_iter().collect(),
+                    options,
+                    aggregate_operation,
+                }
+            }
+            ChangeStreamTarget::Cluster(database) => {
+                let aggregate_target = AggregateTarget::Database(database.name().to_string());
+                doc.get_document_mut("$changeStream")
+                    .unwrap()
+                    .insert("allChangesForCluster", true);
+                aggregate_pipeline.push(doc);
+                aggregate_pipeline.extend(pipeline.clone());
+                let aggregate_operation = Aggregate::new(
+                    aggregate_target,
+                    Vec::new(),
+                    options.clone().map(|opts| opts.get_aggregation_options()),
+                );
+                Self {
+                    target,
+                    pipeline: pipeline.into_iter().collect(),
+                    options,
+                    aggregate_operation,
+                }
+            }
+        }
+    }
+}
+impl Operation for Watch {
+    type O = ChangeStreamSpecification;
+    const NAME: &'static str = "watch";
+
+    fn build(&self, description: &StreamDescription) -> Result<Command> {
+        self.aggregate_operation.build(description)
+    }
+
+    fn handle_response(&self, response: CommandResponse) -> Result<Self::O> {
+        self.aggregate_operation.handle_response(response)?;
```

Comment:
The CursorSpecification needs to be saved in the ChangeStreamSpecification

Reply:
fixed